framework_tvb/tvb/core/neocom/_registry.py

```diff
@@ -57,10 +68,24 @@ def get_h5file_for_datatype(self, datatype_class):
                 return self._h5file_for_datatype[base]
         return H5File
 
-    def get_datatype_for_h5file(self, h5file_class):
+    def get_base_datatype_for_h5file(self, h5file_class):
         # type: (typing.Type[H5File]) -> typing.Type[HasTraits]
         return self._datatype_for_h5file[h5file_class]
 
+    def _determine_proper_datatype_from_h5(self, h5file, subtype):
+        dct = {SurfaceH5: partial(make_surface, subtype),
```

Comment:
Could it work to add 4th optional parameter in `Registry.register_datatype()`
something like `REGISTRY.register_datatype(Surface, SurfaceH5, SurfaceIndex, make_surface)` ?

Reply:
Yes, I changed the code to work as suggested, thanks!